test/integration/crud/explain.test.ts

```diff
@@ -0,0 +1,129 @@
+import { expect } from 'chai';
+import { once } from 'events';
+
+import {
+  type Collection,
+  type CommandStartedEvent,
+  type Db,
+  type MongoClient,
+  MongoServerError
+} from '../../mongodb';
+
+const explain = [true, false, 'queryPlanner', 'allPlansExecution', 'executionStats', 'invalid'];
+
+describe('Explain', function () {
+  let client: MongoClient;
+  let db: Db;
+  let collection: Collection;
+  let commandStartedPromise: Promise<CommandStartedEvent[]>;
+  const ops = [
+    async (explain: boolean | string) => await collection.deleteOne({ a: 1 }, { explain }),
+    async (explain: boolean | string) => await collection.deleteMany({ a: 1 }, { explain }),
+    async (explain: boolean | string) =>
+      await collection.updateOne({ a: 1 }, { $inc: { a: 2 } }, { explain }),
+    async (explain: boolean | string) =>
+      await collection.updateMany({ a: 1 }, { $inc: { a: 2 } }, { explain }),
+    async (explain: boolean | string) => await collection.distinct('a', {}, { explain }),
+    async (explain: boolean | string) => await collection.findOneAndDelete({ a: 1 }, { explain }),
+    async (explain: boolean | string) => await collection.findOne({ a: 1 }, { explain }),
+    (explain: boolean | string) => collection.find({ a: 1 }).explain(explain),
+    async (explain: boolean | string) =>
+      await collection.findOneAndReplace({ a: 1 }, { a: 2 }, { explain }),
+    async (explain: boolean | string) =>
+      await collection
+        .aggregate([{ $project: { a: 1 } }, { $group: { _id: '$a' } }], { explain })
+        .toArray()
+  ];
+
+  const opNames = [
+    'deleteOne',
+    'deleteMany',
+    'updateOne',
+    'updateMany',
+    'distinct',
+    'findOneAndDelete',
+    'findOne',
+    'find',
+    'findOneAndReplace',
+    'aggregate'
+  ];
```

Comment:
For ease of maintenance later on, I think it would be good to have these in a single array shaped like:

```ts
const ops = [
	{name: 'deleteOne', op: ...},
	...
]
```

So if we were to ever add or remove an operation, we wouldn't run into problems with the arrays being different sizes or having an operation be given the wrong name in the test output which could be confusing.

In the loop over the operations, any reference to the name would then reference `op.name` and the operation itself would be `op.op`